src/types.rs

```diff
@@ -825,6 +825,169 @@ impl SourceMap {
 
         Ok((sm, mapping))
     }
+
+    /// Composes two sourcemaps.
+    ///
+    /// This function assumes that `left` only maps between two files and
+    /// its target file is the source file of `right`. In other words, if `left`
+    /// maps from `transformed.js` to `minitfied.js` and `right`
+    /// maps from `minified.js` to `original_1.js`, …, `original_n.js`, then
+    /// the resulting sourcemap maps from `transformed.js`
+    /// to to `original_1.js`, …, `original_n.js`.
+    ///
+    /// The source root, sources, source contents, and names will be copied from `right`. The only information
+    /// that is used from `first` are the mappings.
```

Comment:
If the left sourcemap has multiple target files, this will just silently do the wrong thing. If we wanted to generalize this, we would need a map of sourcemaps on the right, one for each of the left sourcemap's target files. In principle I don't think it's any harder than what we're doing now, just additional effort for functionality we don't (currently) need.

Index maps, as far as I understand, fulfill a slightly different purpose. An index map allows you to concatenate `foo.js` and `bar.js` and say "for this code range (covering `foo.js`) use `foo.js.map`, for the rest (covering `bar.js`) use `bar.js.map`".